yt_dlp/extractor/genius.py

```diff
@@ -0,0 +1,130 @@
+            'brightcove_modal_web_player_id', 'brightcove_song_story_web_player_id', default='S1ZcmcOC1x')
+
+        return self.url_result(
+            smuggle_url(
+                f'https://players.brightcove.net/{account_id}/{player_id}_default/index.html?videoId={video_id}',
+                {'referrer': url}), 'BrightcoveNew', video_id)
+
+
+class GeniusLyricsIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?genius\.com/(?P<id>[^?/#]+)-lyrics[?/#]?'
+    _TESTS = [{
+        'url': 'https://genius.com/Lil-baby-heyy-lyrics',
+        'playlist_mincount': 2,
+        'info_dict': {
+            'id': '8454545',
+            'title': 'Heyy',
+            'description': 'Heyy by Lil Baby',
+        },
+    }, {
+        'url': 'https://genius.com/Outkast-two-dope-boyz-in-a-cadillac-lyrics',
+        'playlist_mincount': 1,
+        'info_dict': {
+            'id': '36239',
+            'title': 'Two Dope Boyz (In a Cadillac)',
+            'description': 'Two Dope Boyz (In a Cadillac) by OutKast',
+        },
+    }, {
+        'url': 'https://genius.com/Playboi-carti-rip-lyrics',
+        'playlist_mincount': 1,
+        'info_dict': {
+            'id': '3710582',
+            'title': 'R.I.P.',
+            'description': 'R.I.P. by Playboi Carti',
+        },
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        escaped_json = self._search_regex(
+            r'window\.__PRELOADED_STATE__\s*=\s*JSON\.parse\((.*)\);', webpage, 'escaped json', display_id)
+        unescaped_json = self._parse_json(escaped_json, display_id, transform_source=js_to_json)
+        song_info = self._parse_json(unescaped_json, display_id)
+        song_id = str_or_none(traverse_obj(song_info, ('songPage', 'song')))
+        if not song_id:
+            raise ExtractorError('Song id not found in webpage')
+
+        title = traverse_obj(
+            song_info, ('songPage', 'trackingData', lambda _, x: x['key'] == 'Title', 'value'),
+            get_all=False, default='untitled')
+        artist = traverse_obj(
+            song_info, ('songPage', 'trackingData', lambda _, x: x['key'] == 'Primary Artist', 'value'),
+            get_all=False, default='unknown artist')
+        media = traverse_obj(
+            song_info, ('entities', 'songs', song_id, 'media'), expected_type=list, default=[])
+
+        entries = []
+        for m in media:
+            if m.get('type') in ('video', 'audio') and m.get('url'):
+                if m.get('provider') == 'spotify':
+                    continue
+                elif m.get('provider') == 'youtube':
+                    entries.append(self.url_result(m['url'], 'Youtube', song_id))
+                elif m.get('provider') == 'soundcloud':
+                    entries.append(self.url_result(m['url'], 'Soundcloud', song_id))
+                else:
+                    entries.append(self.url_result(m['url']))
```

Comment:
Not worth it just to add `ie_key` imo
```suggestion
                else:
                    entries.append(self.url_result(m['url']))
```